lib/FrontendPlugin.cpp

```diff
@@ -550,6 +558,20 @@ struct ExtraValidationConsumer final : public ASTConsumer {
                            "%select{SSBOs|UBOs|push constants}0");
   }
 
+  bool IsLongVectorType(QualType type) {
```

Comment:
Why does this only check pointers or vectors? What if the kernel argument has a long vector in a struct or array? If they are meant to be supported, add a comment to indicate the difference otherwise perhaps just reuse IsSupportedType but provide it an extra argument.

Please also add tests to cover structs and arrays to cover those cases.

Reply:
This is an oversight that I'll address, thanks for bringing it up. I'll add the tests too.

Reply:
When addressing this, I realised `IsSupportedType` didn't cover all cases. The situation should be a bit better now, although there's an `llvm_unreachable` for situation not covered by the existing and additional test-cases.

`IsLongVectorType` has now been removed in favour of using `IsSupportedType`.